src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -1282,6 +1287,12 @@ object Parsers {
 
     /* -------- PATTERNS ------------------------------------------- */
 
+    def scala2WildcardStarMessage1  =
```

Comment:
I'd refactor a little bit differently.

1. Introduce a method 

        def migrationWarningOrError(msg: String, pos: Int) = ...

2. Drop the message defs here